core/src/datamatrix/DMDecoder.cpp

```diff
@@ -158,7 +166,7 @@ static Mode DecodeAsciiSegment(BitSource& bits, std::string& result, std::string
 {
 	Shift128 upperShift;
 
-	while (bits.available() >= 8) {
+	for (int position = 1; bits.available() >= 8; position++) {
```

Comment:
I believe the introduction of this "manual position" counting is not necessary: the `BitSource` class has a member `byteOffset()` that tracks the current position in the bit-stream. It seems we are only reading one byte a time (`bits.readBits(8)`) and you increment position for every byte read. I.e. you can simply use `byteOffset() + 1` where you need to evaluate the position I believe.

Reply:
Excellent, have force-pushed the change [e7ecad8](https://github.com/nu-book/zxing-cpp/commit/e7ecad81948c0efeb55f770a3ce647a413fc6ae8)